src/shared/api/hooks/useBaseMutation.ts

```diff
@@ -0,0 +1,30 @@
+import { MutationFunction, UseMutationOptions, useMutation } from "@tanstack/react-query"
+import { AxiosResponse } from "axios"
+
+import { BaseError } from "../types"
+
+const useBaseMutation = <TRequest, TResponse>(
+  key: string[],
+  mutationFn: MutationFunction<AxiosResponse<TResponse, BaseError>, TRequest>,
+  options?: Omit<
+    UseMutationOptions<AxiosResponse<TResponse, BaseError>, BaseError, TRequest>,
+    "mutationKey" | "mutationFn"
+  >,
+) => {
+  return useMutation(key, mutationFn, {
+    ...options,
+    onError: (error: BaseError, variables: TRequest, context: unknown) => {
+      if (error?.response?.data?.messages?.length) {
+        error.evaluatedMessage = error.response?.data?.messages[0]
```

Comment:
Well, if you have this one then you can make use of it instead of creating useFirstErrorMessage

Reply:
Done